cluster-autoscaler/cloudprovider/gce/gce_manager.go

```diff
@@ -229,6 +247,79 @@ func (m *GceManager) RegisterMig(mig *Mig) bool {
 	return !updated
 }
 
+// UnregisterMig unregisters mig in Gce Manager. Returns true if the node group has been removed.
+func (m *GceManager) UnregisterMig(toBeRemoved *Mig) bool {
+	m.migsMutex.Lock()
+	defer m.migsMutex.Unlock()
+
+	newMigs := make([]*migInformation, 0, len(m.migs))
+	found := false
+	for _, mig := range m.migs {
+		if mig.config.GceRef == toBeRemoved.GceRef {
+			glog.V(4).Infof("Unregistered Mig %s/%s/%s", toBeRemoved.GceRef.Project, toBeRemoved.GceRef.Zone,
+				toBeRemoved.GceRef.Name)
+			found = true
+		} else {
+			newMigs = append(newMigs, mig)
+		}
+	}
+	m.migs = newMigs
+	return found
+}
+
+func (m *GceManager) deleteNodePool(toBeRemoved *Mig) error {
+	m.assertGKE()
+	if !toBeRemoved.Autoprovisioned() {
+		return fmt.Errorf("only autoprovisioned node pools can be deleted")
+	}
+	// TODO: handle multi-zonal node pools.
+	deleteOp, err := m.gkeService.Projects.Zones.Clusters.NodePools.Delete(m.projectId, m.zone, m.clusterName,
+		toBeRemoved.nodePoolName).Do()
+	if err != nil {
+		return err
+	}
+	err = m.waitForGkeOp(deleteOp)
+	if err != nil {
+		return err
+	}
+	return m.fetchAllNodePools()
+}
+
+func (m *GceManager) createNodePool(spec *autoprovisioningSpec) error {
+	m.assertGKE()
+
+	// Todo: handle preemptable
+	// Todo: handle ssd
+	// Todo: handle taints
```

Comment:
s/Todo/TODO 
Pretty sure when people search for TODOs they look for uppercase variant (I certainly do)